examples/platform/esp32/common/CommonDeviceCallbacks.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "CommonDeviceCallbacks.h"
+
+#include "esp_bt.h"
+#include "esp_err.h"
+#include "esp_heap_caps.h"
+#include "esp_log.h"
+#include "esp_nimble_hci.h"
+#include "nimble/nimble_port.h"
+#include "route_hook/esp_route_hook.h"
+#include <app-common/zap-generated/attribute-id.h>
+#include <app-common/zap-generated/cluster-id.h>
+#include <app/server/Dnssd.h>
+#include <app/util/util.h>
+#include <lib/support/CodeUtils.h>
+#if CONFIG_ENABLE_OTA_REQUESTOR
+#include <ota/OTAHelper.h>
+#endif
+
+static const char * TAG = "app-devicecallbacks";
+
+using namespace chip;
+using namespace chip::DeviceLayer;
+using namespace chip::System;
+
+DeviceCallbacksDelegate * appDelegate = nullptr;
+
+void CommonDeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_t arg)
+{
+    switch (event->Type)
+    {
+    case DeviceEventType::kInternetConnectivityChange:
+        OnInternetConnectivityChange(event);
+        break;
+
+    case DeviceEventType::kSessionEstablished:
+        OnSessionEstablished(event);
+        break;
+
+    case DeviceEventType::kCHIPoBLEConnectionEstablished:
+        ESP_LOGI(TAG, "CHIPoBLE connection established");
+        break;
+
+    case DeviceEventType::kCHIPoBLEConnectionClosed:
+        ESP_LOGI(TAG, "CHIPoBLE disconnected");
+        break;
+
+    case DeviceEventType::kCommissioningComplete: {
+        ESP_LOGI(TAG, "Commissioning complete");
```

Comment:
Please print the error code too. Also, perform BLE deinit only when the commissioning is successful (error = 0)?

Reply:
Commissioning complete event comes with FabIndex and PeerNodeId [here](https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/CHIPDeviceEvent.h#L475) ,don’t have error code.  Commissioning complete event comes only when commissioning completed [here](https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/network-commissioning/network-commissioning.cpp#L597-L604).